ethpm_cli/commands/install.py

```diff
@@ -119,6 +129,93 @@ def get_package_aliases(package_name: str, config: Config) -> Iterable[Tuple[str
+
+        connected_chain_id = config.w3.eth.chainId
+        active_registry_uri = parse_registry_uri(active_registry.uri)
+        if not to_int(text=active_registry_uri.chain_id) == connected_chain_id:
+            raise InstallError(
+                f"Registry URI chain: {active_registry_uri.chain_id} doesn't match "
+                f"connected web3: {connected_chain_id}."
+            )
+
+        config.w3.pm.set_registry(active_registry_uri.address)
+        all_pkg_names = config.w3.pm.get_all_package_names()
+        if installed_pkg.resolved_package_name not in all_pkg_names:
+            raise InstallError(
+                f"{installed_pkg.resolved_package_name} is not available on the active registry "
+                f"{active_registry.uri}. Available packages include: {all_pkg_names}."
+            )
+
+        all_release_data = config.w3.pm.get_all_package_releases(
+            installed_pkg.resolved_package_name
+        )
+        all_versions = [version for version, _ in all_release_data]
+
+        if installed_pkg.resolved_version not in all_versions:
+            raise InstallError(
+                f"{installed_pkg.resolved_package_name}@{installed_pkg.resolved_version} not "
+                f"found on the active registry {active_registry.uri}."
+            )
+
+        on_chain_install_uri = pluck_release_data(
+            all_release_data, installed_pkg.resolved_version
+        )
+        if on_chain_install_uri != installed_pkg.resolved_uri:
+            raise InstallError(
+                f"Install URI found on active registry for {installed_pkg.resolved_package_name}@"
+                f"{installed_pkg.resolved_version}: {on_chain_install_uri} does not match the "
+                f"install URI found in local lockfile: {installed_pkg.resolved_uri}."
+            )
+
+        cli_logger.info(
+            f"{len(all_versions)} versions of {installed_pkg.resolved_package_name} "
+            f"found on the active registry: {all_versions}."
+        )
+        while True:
+            target_version = input("Please enter the version you want to install. ")
+            if target_version == installed_pkg.resolved_version:
+                cli_logger.info(f"Version already installed: {target_version}. ")
+            elif target_version not in all_versions:
+                cli_logger.info(f"Version unavailable: {target_version}. ")
+            else:
+                break
+
+        # Create an updated args/Package for new install
+        updated_args = copy.deepcopy(args)
+        if installed_pkg.resolved_package_name != args.package:
+            updated_args.alias = args.package
+        updated_args.uri = pluck_release_data(all_release_data, target_version)
+        updated_args.package_version = target_version
+        updated_package = Package(updated_args, config.ipfs_backend)
+
+        uninstall_package(args.package, config)
+        install_package(updated_package, config)
```

Comment:
I think you need to make this operation and the previous one atomic.  If the program errors out between lines 203/204 then you'll end up in an inconsistent state with the previous package uninstalled but the new one missing.